README.md

```diff
@@ -64,6 +64,116 @@ where:
+
+Deploy istio:
+```bash
+juju add-model istio-system
+juju deploy istio-k8s -m istio-system --channel edge --trust
+juju deploy istio-ingress-k8s -m istio-system --channel edge --trust
+```
+
+Deploy Knative:
+```bash
+juju add-model knative-user
+juju deploy ./charms/knative-operator/knative-operator_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --resource knative-operator-image=gcr.io/knative-releases/knative.dev/operator/cmd/operator@sha256:18e7a0c612efdce11d9d396fa9a1469fa206bdb2a817b5426a595ed57a5c3daf \
+  --resource knative-operator-webhook-image=gcr.io/knative-releases/knative.dev/operator/cmd/webhook@sha256:dc2b23c9f66869a8617a51f2f0af0f56d9dc6a85b71cff587161eb9711aef26a
+juju deploy ./charms/knative-serving/knative-serving_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --config namespace=knative-serving-1 \
+  --config istio.gateway.namespace=istio-system \
+  --config domain.name=10.64.140.43.nip.io \  # <-- may be different if you have a different load balancer configuration
+  --config istio.gateway.name=istio-ingress-k8s \
+  --config version=1.15.0 \
+  --config ingress-class=gateway-api.ingress.networking.knative.dev  # <-- enables the gateway api
+```
+
+Deploy a Knative Service via the [kn CLI](https://knative.dev/docs/client/install-kn/) and use it:
+```bash
+kn service create hello \
+--image ghcr.io/knative/helloworld-go:latest \
+--port 8080 \
+--env TARGET=World
+
+curl http://hello.default.10.64.140.43.nip.io
+# Hello World!
+```
+
+### Using an Istio Gateway alongside the new Gateway API
+
+Istio's implementation of the [Istio Gateway](https://istio.io/latest/docs/reference/config/networking/gateway/) and [Gateway API Gateway](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) both rely on the controlling the same deployment proxy.  Because of this, we can bind an [Istio Gateway](https://istio.io/latest/docs/reference/config/networking/gateway/) to the Kubernetes `deployment` that is created for Istio's [Gateway API Gateway](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway), effectively allowing control over the `Gateway` through both `VirtualServices` and `HTTPRoutes`.
+
+Deploy Istio:
+```bash
+juju add-model istio-system
+juju deploy istio-k8s -m istio-system --channel edge --trust
+juju deploy istio-ingress-k8s -m istio-system --channel edge --trust
+```
+
+Deploy an Istio (old api) Gateway:
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1
+kind: Gateway
+metadata:
+  name: old-gateway-api
+  namespace: istio-system
+spec:
+  selector:
+    # TODO: There might be a better selector to use here
+    gateway.networking.k8s.io/gateway-name: istio-ingress-k8s
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+      # wildcard so we can use it for knative, which adds to the front of the domain
+    - "*.10.64.140.43.nip.io"
```

Comment:
Is the wildcard a prereq here, meaning does the operator by default insert subdomains for workloads it deploys or is it just to add support when needed?, if it's a prereq we will need to do something about the ingress charm to support wildcards as well.

Reply:
Yeah it is a prereq.  All knative urls are `my-knative-service.my-domain`.  So we'll have to support them somehow